### PR TITLE
Improve label and error messages to reduce confusion about test set and external features

### PR DESCRIPTION
**Pull request by alexcombessie**
[ch58081](https://app.clubhouse.io/dataiku/story/58081/improve-labels-to-reduce-confusion-about-test-set-and-external-features)

### CODE REVIEW

**Review by clemencebic — COMMENTED** [review]
Re wording seems ok for me !
I would add a few more indications : 
- forecasting horizon : we could add a small description, something like (number of future timesteps for which you want to predict values)
- when you provide an external features dataset that has not the same size as the forecast horizon, you get the following error message "<class 'ValueError'>: Length of future external features timeseries must be equal to the training prediction length". I would change 'the training prediction length' to 'the forecast horizon', so that the user understands better the link between the settings and the error

**Comment by alexcombessie** [comment]
Great remarks! I have pushed more commits to improve descriptions and error messages

**Review by clemencebic — APPROVED** [review]
👍

File: custom-recipes/timeseries-forecast-1-train-evaluate/recipe.json [hunk 1]
```diff
@@ -29,19 +29,19 @@
...
             "arity": "UNARY",
             "required": true,
             "acceptsDataset": true,
             "acceptsManagedFolder": false
         },
         {
             "name": "evaluation_forecasts_dataset",
-            "label": "Backtest forecasts (optional)",
-            "description": "Backtest data used to compute performance metrics",
+            "label": "Test dataset",
```
Comment (alexbourret) [inline custom-recipes/timeseries-forecast-1-train-evaluate/recipe.json:41]: Would Test results / Test results dataset be clearer ?

File: custom-recipes/timeseries-forecast-1-train-evaluate/recipe.json [hunk 2]
```diff
@@ -183,6 +183,7 @@
             "name": "prediction_length",
             "label": "Forecasting horizon",
             "type": "INT",
+            "description": "Number of time steps to predict future values",
```
Comment (alexbourret) [inline custom-recipes/timeseries-forecast-1-train-evaluate/recipe.json:186]: Number of time steps future values to predict / Number of time units future values to predict ?

**Review by alexbourret — APPROVED** [review]

**Pull request merged**